pages/api/teams/[team]/index.ts

```diff
@@ -2,24 +2,35 @@ import type { NextApiRequest, NextApiResponse } from "next";
 
 import { getSession } from "@lib/auth";
 import prisma from "@lib/prisma";
+import { getTeamWithMembers } from "@lib/queries/teams";
 
 export default async function handler(req: NextApiRequest, res: NextApiResponse) {
   const session = await getSession({ req: req });
   if (!session) {
     return res.status(401).json({ message: "Not authenticated" });
   }
+  if (!session.user?.id) {
+    console.log("Received session token without a user id.");
+    return res.status(500).json({ message: "Something went wrong." });
+  }
+  if (!req.query.team) {
+    console.log("Missing team query param.");
+    return res.status(500).json({ message: "Something went wrong." });
+  }
+
+  const teamId = parseInt(req.query.team as string);
 
+  // GET /api/teams/{team}
+  if (req.method === "GET") {
+    const team = await getTeamWithMembers(teamId);
+    return res.status(200).json({ team });
+  }
```

Comment:
Despite these changes (I made them before converting to tRPC) the teams API section is now redundant. Leaving this here in case it is used by third parties or such.